src/Microsoft.Diagnostics.Monitoring.EventPipe/Triggers/Shared/SharedTriggerImplHelper.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe.Triggers.Shared
+{
+    internal static class SharedTriggerImplHelper
+    {
+        public static bool HasSatisfiedCondition(ref long? latestTicks, ref long? targetTicks, long windowTicks, long intervalTicks, ICounterPayload payload, bool passesValueFilter)
```

Comment:
Consider renaming this method. HasSatisfiedCondition has a lot of side effects to its parameters that might not be obvious.